pyiron_base/server/generic.py

```diff
@@ -246,6 +247,12 @@ def cores(self, new_cores):
         Args:
             new_cores (int): number of cores
         """
+        if not isinstance(new_cores, numbers.Integral):
```

Comment:
What do we need this line for?

Reply:
I meant the very line I commented on. I still don’t see what we need that line for

Reply:
If it's already an integer, no need to try a conversion.  We could remove the line and add a check in case the type casting fails.

Reply:
Ok then I would have done

```python
if new_cores != int(new_coes):
    raise ValueError(f"cores must be an integer number, not {new_cores}!")
new_cores = int(new_cores)
```

This would have spared two lines. But I'm really super unhappy with the fact that it was merged while it was unresolved, especially my comment was pertinent.

Now you don't need to open another PR just to change this.

Reply:
Yup, sorry for that.  I did not think it was a critical issue and wanted to have the fix quickly available.